libafl/src/executors/inprocess.rs

```diff
@@ -696,6 +702,118 @@ mod windows_exception_handler {
     }
 }
 
+#[cfg(all(feature = "std", unix))]
+pub struct InProcessForkExecutor<'a, H, I, OT, S>
+where
+    H: FnMut(&I) -> ExitKind,
+    I: Input,
+    OT: ObserversTuple<I, S>,
+{
+    harness_fn: &'a mut H,
+    observers: OT,
+    phantom: PhantomData<(I, S)>,
+}
+
+#[cfg(all(feature = "std", unix))]
+impl<'a, EM, H, I, OT, S, Z> Executor<EM, I, S, Z> for InProcessForkExecutor<'a, H, I, OT, S>
+where
+    H: FnMut(&I) -> ExitKind,
+    I: Input,
+    OT: ObserversTuple<I, S>,
+{
+    #[allow(unreachable_code)]
+    #[inline]
+    fn run_target(
+        &mut self,
+        _fuzzer: &mut Z,
+        _state: &mut S,
+        _mgr: &mut EM,
+        input: &I,
+    ) -> Result<ExitKind, Error> {
+        unsafe {
+            match fork() {
+                Ok(ForkResult::Child) => {
```

Comment:
after fork we'll probably have to call `post_fork` on the shmem provider, else MacOS and Android won't work